File: problem08-containerWithWater/containerWithWater.js
/**
 * @name containerWithWater
 * @desc For an array of positive integer, where each integer represents the height of a vertical line. Return the area of water it would hold.
 * @param arr - Array that contains positive integer
 */
function containerWithWater(arr) {
  // intialize the variables
  let maxArea = 0;
  // iterate through the array
  for (let i = 0; i < arr.length; i++) {
    for (let j = 1; j < arr.length + 1; j++) {
      const height = Math.min(arr[i], arr[j]);
      const width = j - i;
      const area = height * width;
      if (area > maxArea) {
        maxArea = area;
      }
    }
  }
  return maxArea;
}
const heightsArray = [7, 1, 2, 3, 9];
const area = containerWithWater(heightsArray);
console.log(area);

File: problem08-containerWithWater/containerWithWater.py
'''
@name containerWithWater 
@desc For an array of positive integer, where each integer represents the height of a vertical line. Return the area of water it would hold.
@param arr - Array that contains positive integers 
@return maxArea - The max area that can be calculated 
'''
def containerWithWater(arr):
    maxArea = 0
    for i in range(len(arr)):
        for j in range(1, len(arr)):
            height = min(int(arr[i]), int(arr[j]))
            width = j - i 
            area = height * width
            if (area > maxArea):
                maxArea = area
    return maxArea 
heightsArray = [7, 1, 2, 3, 9]
print(containerWithWater(heightsArray))